nomos-services/time/src/lib.rs

```diff
@@ -119,24 +123,10 @@ where
         loop {
             tokio::select! {
                 Some(service_message) = inbound_relay.recv() => {
-                    match service_message {
-                        TimeServiceMessage::Subscribe { sender} => {
-                            let channel_stream = BroadcastStream::new(broadcast_receiver.resubscribe()).filter_map(|r| Box::pin(async {match r {
-                                Ok(tick) => Some(tick),
-                                Err(e) => {
-                                    // log lagging errors, services should always aim to be ready for next slot
-                                    error!("Lagging behind slot ticks: {e:?}");
-                                    None
-                                }
-                            }}));
-                            let stream = Pin::new(Box::new(channel_stream));
-                            if let Err(_e) = sender.send(stream) {
-                                error!("Error subscribing to time event: Couldn't send back a response");
-                            }
-                        }
-                    }
+                    handle_service_message(service_message, &broadcast_receiver, &current_slot_tick);
                 }
                 Some(slot_tick) = tick_stream.next() => {
+                    current_slot_tick = slot_tick;
```

Comment:
I decided to maintain the `current_slot_tick` in this layer (not inside each backend). I tried the other way but realized that this way is much simpler/cleaner. For example, it's not simple to maintain the `current_slot_tick` in the NTP backend because it calibrates the current time and replace the stream of future ticks periodically.